manifests/ssltools/nssdb.pp

```diff
@@ -1,50 +1,27 @@
 # Sets up nssdb
 class certs::ssltools::nssdb (
   $nss_db_dir = $certs::nss_db_dir,
+  $nss_db_password_path = $certs::nss_db_password_path,
+  $nss_db_password = $certs::nss_db_password,
   $group = 'qpidd',
 )  {
   Exec { logoutput => 'on_failure' }
 
-  $nss_db_password_file   = "${nss_db_dir}/nss_db_password-file"
-
-  $nssdb_files = $facts['os']['release']['major'] ? {
-    '7' => ["${nss_db_dir}/cert8.db", "${nss_db_dir}/key3.db", "${nss_db_dir}/secmod.db"],
-    default => ["${nss_db_dir}/cert9.db", "${nss_db_dir}/key4.db", "${nss_db_dir}/pkcs11.txt"]
-  }
-
   ensure_packages(['openssl', 'nss-tools'])
 
-  file { $nss_db_dir:
-    ensure => directory,
-    owner  => 'root',
-    group  => $group,
-    mode   => '0755',
-  } ->
-  exec { 'generate-nss-password':
-    command => "openssl rand -base64 24 > ${nss_db_password_file}",
-    path    => '/usr/bin',
-    umask   => '0027',
+  file { $nss_db_password_path:
+    ensure  => file,
+    content => $nss_db_password,
+    owner   => 'root',
     group   => $group,
-    creates => $nss_db_password_file,
-    require => Package['openssl'],
-  } ->
-  file { $nss_db_password_file:
-    ensure => file,
-    owner  => 'root',
-    group  => $group,
-    mode   => '0640',
-  } ->
-  exec { 'create-nss-db':
-    command => "certutil -N -d '${nss_db_dir}' -f '${nss_db_password_file}'",
-    path    => '/usr/bin',
-    umask   => '0027',
-    group   => $group,
-    creates => $nssdb_files,
-    require => Package['nss-tools'],
-  } ->
-  file { $nssdb_files:
-    owner => 'root',
-    group => $group,
-    mode  => '0640',
+    mode    => '0640',
+  }
+
+  nssdb { $nss_db_dir:
+    ensure        => present,
+    password_file => $nss_db_password_path,
```

Comment:
'Tis a bit ironic that we moved to using `_path` terminology as the variable but are passing it to a parameter using `_file` terminology. This gets into some bikeshedding of the naming conventions but I think it somewhat draws to my point here https://github.com/theforeman/puppet-certs/pull/344/files#r655487885

I think we should strive for consistency in our use of words.

Reply:
Many tools actually have some option that refers to file. For example, `-storepass:file` or `--password file:/path/to/file`. Given nss and certutil both refer to a file and not a path, I think the original password_file naming is not something we need to change.

Reply:
I am really just looking for consistency, because we already do use this convention at https://github.com/theforeman/puppet-certs/blob/fe831c38c8488be70ca28bb41dfb24854d94a8a6/manifests/candlepin.pp#L73-L74

I think @ehelms shares the same goal and I suspect you are on the same page also, so I think it comes down to a matter of what should we address in this PR vs. what is well beyond the scope and needs to be addressed separately.